File: backend/src/main/java/com/foureyes/moai/backend/domain/schedule/entity/Schedule.java
package com.foureyes.moai.backend.domain.schedule.entity;

import com.foureyes.moai.backend.domain.study.entity.StudyGroup;
import com.foureyes.moai.backend.domain.user.entity.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "schedules")
public class Schedule {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "study_id")
    private StudyGroup studyGroup;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "start_datetime", nullable = false)
    private LocalDateTime startDatetime;

    @Column(name = "end_datetime", nullable = false)
    private LocalDateTime endDatetime;

    @Column(nullable = false)
    private String title;

    @Column
    private String memo;
}

File: frontend/src/types/schedule.ts
export interface ScheduleItem {
  id: number
  startDatetime: string
  endDatetime: string
  title: string
  memo?: string
}

export interface MyScheduleItem extends ScheduleItem {
  name: string
  description: string
  image: string
}

export interface CreateScheduleReq {
  studyId: number
  userId: number
  startDatetime: string
  endDatetime: string
  title: string
  memo?: string
}

export interface EditScheduleReq {
  id: number
  studyId: number
  userId: number
  startDatetime?: string
  endDatetime?: string
  title?: string
  memo?: string
}

export interface DeleteScheduleReq {
  user_id: number
}
